internal/adapters/cloud/aws/s3/s3_test.go

```diff
@@ -338,7 +338,6 @@ func bootstrapBucketEncryption(t *testing.T, api *s3api.Client, ctx context.Cont
 					ApplyServerSideEncryptionByDefault: &s3types.ServerSideEncryptionByDefault{
 						SSEAlgorithm: s3types.ServerSideEncryptionAes256,
 					},
-					BucketKeyEnabled: true,
```

Comment:
I would keep this in, as it is another way for a bucket to be encrypted. As for the test, if you implement the suggestion I mentioned above, the test should be green for the cases when either `BucketKeyEnabled` and / or `ApplyServerSideEncryptionByDefault` is set.

Ideally they should be two different cases but I think we can leave that for now if you wish.